bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -426,7 +426,7 @@ public BookieImpl(ServerConfiguration conf,
         }
 
         // instantiate the journals
-        journals = Lists.newArrayList();
+        journals = Lists.newArrayListWithCapacity(journalDirectories.size());
```

Comment:
I think `new ArrayList<>(size)` is better. Guava also recommend jdk8+ to use `new ArrayList<>()`